src/main/java/com/trainigcenter/springtask/persistence/ActorDao.java

```diff
@@ -0,0 +1,23 @@
+package com.trainigcenter.springtask.persistence;
+
+import com.trainigcenter.springtask.domain.Actor;
+
+import java.util.List;
+
+public interface ActorDao {
+
+    Actor findById(Integer id);
+
+    List<Actor> findAll();
+
+    Actor findByName(String name);
+
+    Actor add(Actor actor);
+
+    Actor update(Actor actor);
+
+    void delete(Actor actor);
+
+    Actor findByIdWithMovies(Integer id);
```

Comment:
I would prefer to return Optional<Actor> from methods when I cannot guarantee that the object really exists.